src/libcore/ptr.rs

```diff
@@ -93,17 +216,29 @@ pub const fn null_mut<T>() -> *mut T { 0 as *mut T }
 /// Swaps the values at two mutable locations of the same type, without
 /// deinitializing either.
 ///
-/// The values pointed at by `x` and `y` may overlap, unlike `mem::swap` which
-/// is otherwise equivalent. If the values do overlap, then the overlapping
-/// region of memory from `x` will be used. This is demonstrated in the
-/// examples section below.
+/// But for the following two exceptions, this function is semantically
+/// equivalent to [`mem::swap`]:
+///
+/// * It operates on raw pointers instead of references. When references are
+///   available, [`mem::swap`] should be preferred.
+///
+/// * The two pointed-to values may overlap. If the values do overlap, then the
```

Comment:
I know that's pre-existing, but this is incomplete.

For example, the example claims that:
```Rust
/// let mut array = [0, 1, 2, 3];
///
/// let x = array[0..].as_mut_ptr() as *mut [u32; 3];
/// let y = array[1..].as_mut_ptr() as *mut [u32; 3];
///
/// unsafe {
///     ptr::swap(x, y);
///     assert_eq!([1, 0, 1, 2], array);
/// }
```

that's what you get by writing to `x` first, then writing to `y`. If you wrote to `y` first, then overwrote that with `x`, you'll get `[1, 2, 3, 2]`. Do we make any guarantees about which option you'll get?

Reply:
Good question. I wouldn't know. Should I just document the current behavior?

Reply:
I feel that would be adequate, but I would ask someone on the libs team for how much that is intended.

Reply:
Hm actually I think @arielb1 what you are saying is exactly what the docs mean when they say "the overlapping region of memory from `x` will be used". So I am not sure what you are suggesting should be changed here. Or are you just asking for a better wording? Not sure how to put that though.

Reply:
As I said, there are at least 2 possible different outputs depending on how `swap` is implemented. What happens should be documented somehow.

Reply:
The docs say which one it is: The one where `x` is used for the overlapping region. That's why the overlapping region is `0, 1` in the example, nor `2, 3`.